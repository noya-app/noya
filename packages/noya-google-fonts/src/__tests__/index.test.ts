import { GoogleFontProvider } from '../index';

const {
  getFontFamilyIdList,
  getFontFamilyId,
  getFontDescriptorsForFamily,
  getFontFileUrl,
} = GoogleFontProvider;

const robotoFamilyID = getFontFamilyId('roboto')!;

test('it should get font id', () => {
  expect(getFontFamilyId('Noto Sans')?.toString()).toEqual('notosans');
  expect(getFontFamilyId('NotoSans')?.toString()).toEqual('notosans');
  expect(getFontFamilyId('Noto-Sans')?.toString()).toEqual('notosans');
  expect(getFontFamilyId('Noto_Sans')?.toString()).toEqual('notosans');
});

test('it should get all font family ids', () => {
  expect(getFontFamilyIdList().length).toEqual(1075);
});

test('it should get font variants', () => {
  expect(getFontDescriptorsForFamily(robotoFamilyID)).toMatchSnapshot();
});

test('it should get font file', () => {
  expect(
    getFontFileUrl({
      fontFamilyId: robotoFamilyID,
      fontSlant: 'upright',
      fontWeight: 'bold',
    }),
  ).toEqual(
    'https://fonts.gstatic.com/s/roboto/v27/KFOlCnqEu92Fr1MmWUlvAx05IsDqlA.ttf',
  );
});
