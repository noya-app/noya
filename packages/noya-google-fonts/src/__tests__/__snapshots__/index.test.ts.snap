// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should get font variants 1`] = `
Array [
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "ultralight",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "ultralight",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "light",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "light",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "regular",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "regular",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "medium",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "medium",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "bold",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "bold",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "upright",
    "fontWeight": "black",
  },
  Object {
    "fontFamilyId": "roboto",
    "fontSlant": "italic",
    "fontWeight": "black",
  },
]
`;
